### ISSUE TITLE
False positive for 933100 for <?xpacket

### ISSUE THREAD

**Issue by MarcelEeken** [coreruleset/coreruleset#4443]
### Description

There seems to be a false positive in rule 933100 with regards to [Extensible Metadata Platform](https://en.wikipedia.org/wiki/Extensible_Metadata_Platform).

This uses the `<?xpacket` tag to start and end a metadata segment. Unfortunately this tag is picked up by the PHP regex as a PHP opening tag.

### How to reproduce the misbehavior (-> curl call)

Example curl request:
```curl
curl -X GET 'https://sandbox.coreruleset.org/?test=<?xpacket' \
  --header 'Accept: */*' \
  --header 'x-format-output: txt-matched-rules'
```

### Logs

Example pdf file (Generated through [Canva](https://www.canva.com)) containing this meta tag:

[false_positive_993100.pdf](https://github.com/user-attachments/files/25096323/false_positive_993100.pdf)

### Your Environment

* CRS version: v3.3.2:
* Paranoia level setting (e.g. PL1) :
* Google Cloud Armor

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by fzipi** [comment]
Hi @MarcelEeken ! Thanks for your report.

Unfortunately, you are using a > 5 years old ruleset, and I don't think we can help you with Google Cloud Armor.

But maybe the report works for something in our latest version, 4.23.0.

We'll keep you posted.

**Comment by MarcelEeken** [comment]
I saw indeed that cloud armor is using a really old version unfortunately. 

But if I correctly the current version (4.23.0) is also generating a false positive on this specific tag.

**Comment by touchweb-vincent** [comment]
Hello,

Yes, the current version is also affected.

@fzipi , I can take care of this if you’re okay with it. It would require changing this regex:

https://github.com/coreruleset/coreruleset/blob/main/rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf#L47C71-L47C165

to:

@rx (?i)<\?(?:[^x]|x(?:[^mp]|m(?:[^l]|l(?:[^\s\x0b]|[\s\x0b]+[^a-z]|$)))|$|php)|\[[/\x5c]?php\]

I don’t see any no-go regarding this change within the scope of PHP injection detection.

**Comment by fzipi** [comment]
Thanks @MarcelEeken, and thanks @touchweb-vincent for offering to update the regexp.

Could I ask you also to create a regex-assembly file from it? If this is too much, then at least add as many comments as you can here in the issue, and then we can team up to generate the file.

**Comment by touchweb-vincent** [comment]
With pleasure. Do you have an example of an existing .ra file that implements this kind of pattern: x([^m]|m[^l]) ?

**Comment by fzipi** [comment]
🤷 Maybe?

**Comment by fzipi** [comment]
This one? https://github.com/coreruleset/coreruleset/blob/main/regex-assembly/933100.ra

**Comment by touchweb-vincent** [comment]
Thanks @fzipi , I was expecting something trickier than this.PR done.
